src/modules/template-details/TemplateDetails.tsx

```diff
@@ -1,4 +1,4 @@
-import React, { useEffect, useCallback, useState, useMemo } from 'react';
+import React, { useEffect, useCallback, useState, useMemo, useContext } from 'react';
```

Comment:
Avoid using useContext directly. We using Provider wrap over all context and adding useNameOfProvider hook - check for reference useTechnologiesProvider or usePatternProvider and repeat this pattern

Reply:
done